Electron v8.0.x is out

They've bumped the version of Electron yet again (8.0.1) and I'd like to try it out.  It looks like as before, the DeepSpeech node module will need to create the *electron-v8.0* directory in:

*deepspeech/lib/binding/v0.6.1/darwin-x64/*

```
Error: Cannot find module '/Users/dstein/dev/jaxcore/deepspeech-plugin/node_modules/deepspeech/lib/binding/v0.6.1/darwin-x64/electron-v8.0/deepspeech.node'
[1] Require stack:
[1] - /Users/dstein/dev/jaxcore/deepspeech-plugin/node_modules/deepspeech/index.js
[1] - /Users/dstein/dev/jaxcore/deepspeech-plugin/lib/deepspeech-process.js
[1]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:798:15)
[1]     at Module._load (internal/modules/cjs/loader.js:691:27)
[1]     at Module._load (electron/js2c/asar.js:717:26)
[1]     at Function.Module._load (electron/js2c/asar.js:717:26)
[1]     at Module.require (internal/modules/cjs/loader.js:853:19)
[1]     at require (internal/modules/cjs/helpers.js:74:18)
[1]     at Object.<anonymous> (/Users/dstein/dev/jaxcore/deepspeech-plugin/node_modules/deepspeech/index.js:17:17)
[1]     at Module._compile (internal/modules/cjs/loader.js:968:30)
[1]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:986:10)
[1]     at Module.load (internal/modules/cjs/loader.js:816:32) {
[1]   code: 'MODULE_NOT_FOUND',
[1]   requireStack: [
[1]     '/Users/dstein/dev/jaxcore/deepspeech-plugin/node_modules/deepspeech/index.js',
[1]     '/Users/dstein/dev/jaxcore/deepspeech-plugin/lib/deepspeech-process.js'
[1]   ]
[1] }
```

I tried updating the version of node-abi, but it's still resolving to a non-existent directory -- the same thing that happened in the other Electron versions.

I couldn't quite figure out how adjust the DeepSpeech scripts to do this upgrade myself.

I'm not sure if I'll be able to figure this out on my own.  I don't think I even have swig installed correctly, I downloaded the binary from that taskcluster link, but I get these make errors.

```
swig -c++ -javascript -node deepspeech.i
:1: Error: Unable to find 'swig.swg'
:3: Error: Unable to find 'javascript.swg'
deepspeech.i:2: Error: Unable to find 'typemaps.i'
make: *** [deepspeech_wrap.cxx] Error 1
```

> I'm not sure if I'll be able to figure this out on my own. I don't think I even have swig installed correctly, I downloaded the binary from that taskcluster link, but I get these make errors.
> 
> ```
> swig -c++ -javascript -node deepspeech.i
> :1: Error: Unable to find 'swig.swg'
> :3: Error: Unable to find 'javascript.swg'
> deepspeech.i:2: Error: Unable to find 'typemaps.i'
> make: *** [deepspeech_wrap.cxx] Error 1
> ```

Good news, it seems you can get that to work easily:
```
$ SWIG_LIB=/home/alex/tmp/deepspeech/SWIG/install/share/swig/4.0.2 PATH=$HOME/tmp/deepspeech/SWIG/install/bin/:$HOME/node_modules/.bin/:$PATH make -C native_client/javascript/ NODE_ABI_TARGET=--target=8.0.1 NODE_DIST_URL=--disturl=https://electronjs.org/headers NODE_RUNTIME=--runtime=electron clean node-wrapper npm-pack 
```

> I'm not sure if I'll be able to figure this out on my own. I don't think I even have swig installed correctly, I downloaded the binary from that taskcluster link, but I get these make errors.
> 
> ```
> swig -c++ -javascript -node deepspeech.i
> :1: Error: Unable to find 'swig.swg'
> :3: Error: Unable to find 'javascript.swg'
> deepspeech.i:2: Error: Unable to find 'typemaps.i'
> make: *** [deepspeech_wrap.cxx] Error 1
> ```

There should be enough doc in https://github.com/mozilla/DeepSpeech/blob/master/native_client/README.rst#install-nodejs--electronjs-bindings but it's very fresh, so please advise if it needs improvements.

FTR, I just tested and it looks like it should build / run easily, so I can make a PR for that later today, but that won't cover 0.6.1, you will have anyway to do it on your side. We are trying to make that as easy as we can, please mind the gap between the code and the binary.

--- a/taskcluster/.shared.yml
+++ b/taskcluster/.shared.yml
@@ -60,8 +60,8 @@ cpp:
     env: 'export EXTRA_ENV="DYLD_LIBRARY_PATH=$LOCAL_HOMEBREW_DIRECTORY/lib/:$DYLD_LIBRARY_PATH"'
 system:
   node_gyp_cache:
-    url: 'https://community-tc.services.mozilla.com/api/index/v1/task/project.deepspeech.node-gyp-cache.1/artifacts/public/node-gyp-cache.tar.gz'
-    namespace: 'project.deepspeech.node-gyp-cache.1'
+    url: 'https://community-tc.services.mozilla.com/api/index/v1/task/project.deepspeech.node-gyp-cache.2/artifacts/public/node-gyp-cache.tar.gz'
+    namespace: 'project.deepspeech.node-gyp-cache.2'
   swig:
     repo: "https://github.com/lissyx/swig"
     sha1: "b5fea54d39832d1d132d7dd921b69c0c2c9d5118"

--- a/taskcluster/tc-all-vars.sh
+++ b/taskcluster/tc-all-vars.sh
@@ -82,4 +82,4 @@ SUPPORTED_PYTHON_VERSIONS=${SUPPORTED_PYTHON_VERSIONS:-3.5.8:ucs4 3.6.10:ucs4 3.
 # deepspeech.node-gyp-cache.<X> in both `system.node_gyp_cache` (taskcluster/.shared.yml)
 # and route index (taskcluster/node-gyp-cache.yml) to ensure the cache is updated
 SUPPORTED_NODEJS_VERSIONS=${SUPPORTED_NODEJS_VERSIONS:-10.18.1 11.15.0 12.8.1 13.1.0}
-SUPPORTED_ELECTRONJS_VERSIONS=${SUPPORTED_ELECTRONJS_VERSIONS:-5.0.13 6.0.12 6.1.7 7.0.1 7.1.8}
+SUPPORTED_ELECTRONJS_VERSIONS=${SUPPORTED_ELECTRONJS_VERSIONS:-5.0.13 6.0.12 6.1.7 7.0.1 7.1.8 8.0.1}

--- a/taskcluster/test-electronjs_v8.0-darwin-amd64-opt.yml
+++ b/taskcluster/test-electronjs_v8.0-darwin-amd64-opt.yml
@@ -1,0 +1,14 @@
+build:
+  template_file: test-darwin-opt-base.tyml
+  dependencies:
+    - "darwin-amd64-cpu-opt"
+    - "test-training_16k-linux-amd64-py36m-opt"
+  test_model_task: "test-training_16k-linux-amd64-py36m-opt"
+  system_setup:
+    >
+      ${nodejs.brew.setup} && ${nodejs.brew.prep_12} && ${nodejs.brew.env}
+  args:
+    tests_cmdline: "$TASKCLUSTER_TASK_DIR/DeepSpeech/ds/taskcluster/tc-electron-tests.sh 12.x 8.0.1 16k"
+  metadata:
+    name: "DeepSpeech OSX AMD64 CPU ElectronJS v8.0 tests"
+    description: "Testing DeepSpeech for OSX/AMD64 on ElectronJS v8.0, CPU only, optimized version"

--- a/taskcluster/test-electronjs_v8.0-win-amd64-opt.yml
+++ b/taskcluster/test-electronjs_v8.0-win-amd64-opt.yml
@@ -1,0 +1,14 @@
+build:
+  template_file: test-win-opt-base.tyml
+  dependencies:
+    - "win-amd64-cpu-opt"
+    - "test-training_16k-linux-amd64-py36m-opt"
+  test_model_task: "test-training_16k-linux-amd64-py36m-opt"
+  system_setup:
+    >
+      ${system.sox_win} && ${nodejs.win.prep_12}
+  args:
+    tests_cmdline: "${system.homedir.win}/DeepSpeech/ds/taskcluster/tc-electron-tests.sh 12.x 8.0.1 16k"
+  metadata:
+    name: "DeepSpeech Windows AMD64 CPU ElectronJS v8.0 tests"
+    description: "Testing DeepSpeech for Windows/AMD64 on ElectronJS v8.0, CPU only, optimized version"

--- a/taskcluster/test-electronjs_v8.0_16k-linux-amd64-opt.yml
+++ b/taskcluster/test-electronjs_v8.0_16k-linux-amd64-opt.yml
@@ -1,0 +1,15 @@
+build:
+  template_file: test-linux-opt-base.tyml
+  docker_image: "ubuntu:16.04"
+  dependencies:
+    - "linux-amd64-cpu-opt"
+    - "test-training_16k-linux-amd64-py36m-opt"
+  test_model_task: "test-training_16k-linux-amd64-py36m-opt"
+  system_setup:
+    >
+      ${nodejs.packages_xenial.prep_12} && ${nodejs.packages_xenial.apt_pinning} && apt-get -qq update && apt-get -qq -y install ${nodejs.packages_xenial.apt} ${electronjs.packages_xenial.apt}
+  args:
+    tests_cmdline: "${system.homedir.linux}/DeepSpeech/ds/taskcluster/tc-electron-tests.sh 12.x 8.0.1 16k"
+  metadata:
+    name: "DeepSpeech Linux AMD64 CPU ElectronJS v8.0 tests (16kHz)"
+    description: "Testing DeepSpeech for Linux/AMD64 on ElectronJS v8.0, CPU only, optimized version (16kHz)"

--- a/taskcluster/test-electronjs_v8.0_8k-linux-amd64-opt.yml
+++ b/taskcluster/test-electronjs_v8.0_8k-linux-amd64-opt.yml
@@ -1,0 +1,15 @@
+build:
+  template_file: test-linux-opt-base.tyml
+  docker_image: "ubuntu:16.04"
+  dependencies:
+    - "linux-amd64-cpu-opt"
+    - "test-training_8k-linux-amd64-py36m-opt"
+  test_model_task: "test-training_8k-linux-amd64-py36m-opt"
+  system_setup:
+    >
+      ${nodejs.packages_xenial.prep_12} && ${nodejs.packages_xenial.apt_pinning} && apt-get -qq update && apt-get -qq -y install ${nodejs.packages_xenial.apt} ${electronjs.packages_xenial.apt}
+  args:
+    tests_cmdline: "${system.homedir.linux}/DeepSpeech/ds/taskcluster/tc-electron-tests.sh 12.x 8.0.1 8k"
+  metadata:
+    name: "DeepSpeech Linux AMD64 CPU ElectronJS v8.0 tests (8kHz)"
+    description: "Testing DeepSpeech for Linux/AMD64 on ElectronJS v8.0, CPU only, optimized version (8kHz)"

